crates/oxc_transformer/src/es2022/class_properties/supers.rs

```diff
@@ -34,9 +34,7 @@ impl<'a, 'ctx> ClassProperties<'a, 'ctx> {
         is_callee: bool,
         ctx: &mut TraverseCtx<'a>,
     ) -> Expression<'a> {
-        let property = &member.property;
-        let property =
-            ctx.ast.expression_string_literal(property.span, property.name.clone(), None);
+        let property = Expression::StringLiteral(ctx.ast.alloc(member.property.clone().into()));
```

Comment:
This is shorter, but I'm not really a fan of `impl From<IdentifierName> for StringLiteral`.

Reason is: You're dealing with an owned type on the stack. Usually, it's better to allocate it into the arena as quickly as possible, and then only hold `Box<T>` (8 bytes), references `&T` or `&mut T` (8 bytes), enum types e.g. `Expression`, `Statement` etc (16 bytes), or `Atom`s (16 bytes).

Generally, if you find yourself having to use `ctx.ast.alloc`, you're not in an optimal situation, because it means you have a large type on the stack.

In the original version, you only had to clone the `Atom` (which is essentially a `Copy`), whereas in this new version you have to clone the whole `IdentifierName` to be able to pass it to `from`.

In this case, it's not a big deal, because `IdentifierName` is only 24 bytes, but in general in my opinion it's not the best pattern.

If you wanted to, you could implement this conversion instead as a method on `AstBuilder` - `AstBuilder:: expression_string_literal_from_identifier_name`. But personally I think that'd be overkill unless we need to do that conversion in many places.